src/utils/src/types/team-types.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { ProjectPreview } from './project-types';
+import { User } from './user-types';
+
+export interface Team {
+  teamId: number;
+  teamName: string;
+  leader: User;
+  members: User[];
+  projects: ProjectPreview[];
+}
+
+export interface TeamPreview {
```

Comment:
Not sure how lovely / shitty it is, but I think we might maybe actually be able to get away with using TypeScript Utility types: Pick<Team, "teamId", "teamName"> or a different one / implementation if I'm forgetting

Reply:
ooh cool